src/main/java/com/google/sps/servlets/RouteImage.java

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
+import com.google.sps.data.Images;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+
+@SuppressWarnings("serial")
+public class RouteImage extends HttpServlet {
+  /** Store route's image. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+      String routeId = request.getParameter("name-route");
```

Comment:
The server should check that the current user is allowed to edit the route before saving the uploaded file or updating the route entity.